src/main/java/io/vertx/codegen/annotations/DataObject.java

```diff
@@ -82,11 +82,18 @@
   Class<? extends Case> jsonPropertyNameFormatter() default LowerCamelCase.class;
 
   /**
-   * Returns the expected Base64 (RFC 4648) alphabet to be used. The default is {@code false}. This will use vert.x
-   * default alphabet, (RFC 4648 Table 2) also known as {@code base64url}.
+   * Returns the expected Base64 (RFC 4648) type to be used. The default is {@code "system"}. This is
+   * vert.x default alphabet, usually (RFC 4648 Table 2) also known as {@code base64url}.
    *
-   * @return {@code true} if generated converters for buffers should default to base64 basic encoding
-   * (RFC 4648 Table 1), else base64url (RFC 4648 Table 2).
+   * Allowed values are:
+   *
+   * <ul>
+   *   <li>{@code "system"} - Use the default type as defined in: {@link io.vertx.core.json.impl.JsonUtil}</li>
```

Comment:
do we have a better name than system ?

Reply:
`default`?

Reply:
This name is tricky. It is a `compatibility` feature we have in:

https://github.com/eclipse-vertx/vert.x/blob/master/src/main/java/io/vertx/core/json/impl/JsonUtil.java#L37-L50

It only exists to reduce breakages for users that still deploy vertx 3 and vertx 4 next to each other.

An alternative is to **not** have this mode listed. In this case we can change the logic to:

If type is: `basic` or `base64url` then it's fixed. For undefined values (fallback to the default system wide format as we do currently).

Is this an idea?

Reply:
it works for me